doc/whatsnew/fragments/9210.bugfix

```diff
@@ -0,0 +1,4 @@
+Fix bug for not being able to walk through files when work on `./` as input
+at a directory with a `__init__.py` file.
```

Comment:
This sentence is not really understandable. I'd like to suggest something to improve but I don't even really know what it is meant to say.

Reply:
Sorry I'm not English native. What I mean here is that:

- It's a bug fix
- The current situation is that, if there is a `__init__.py` file in a directory, pylint is not able to walk (iteratre through) the files in this directory